ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/home/view/HomeView.kt

```diff
@@ -189,33 +209,160 @@ private fun HomeMainContent(
                 )
             }
         }
-        Status(walletSnapshot)
+
+        Status(walletSnapshot, updateAvailable)
+
+        Spacer(modifier = Modifier.height(24.dp))
+
         PrimaryButton(onClick = goSend, text = stringResource(R.string.home_button_send))
+
         TertiaryButton(onClick = goRequest, text = stringResource(R.string.home_button_request))
+
         History(transactionHistory)
     }
 }
 
 @Composable
-private fun Status(walletSnapshot: WalletSnapshot) {
-    Column(Modifier.fillMaxWidth()) {
-        Header(text = walletSnapshot.totalBalance().toZecString())
-        Body(
-            text = stringResource(
-                id = R.string.home_status_shielding_format,
-                walletSnapshot.saplingBalance.total.toZecString()
+@Suppress("LongMethod", "MagicNumber")
+private fun Status(walletSnapshot: WalletSnapshot, updateAvailable: Boolean) {
+    // TODO this will go away before PR merged
+    Twig.info { "WALLET: $walletSnapshot" }
+
+    val configuration = LocalConfiguration.current
+    val contentSizeRatioRatio = if (configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        0.45f
+    } else {
+        0.9f
+    }
+
+    // parts sizes
+    val progressCircleStroke = 12.dp
+    val progressCirclePadding = progressCircleStroke + 6.dp
+    val contentPadding = progressCircleStroke + progressCirclePadding + 10.dp
+
+    // parts values
+    var progressCirclePercentage = 0
+    val zecAmountText = walletSnapshot.totalBalance().toZecString()
+    var usdAmountText = walletSnapshot.spendableBalance().toUsdString()
```

Comment:
I think we might need a sealed class to represent currency conversion, as there are at least a few cases:
 - Current conversion rate
 - Stale conversion rate
 - Unable to fetch

Any other cases that you can think of?

Reply:
`usdAmountText` is `var`, because we change its value from just e.g. _$123_ to _$123 so far_ in case of PREPARING, DOWNLOADING or VALIDATING Synchronizer states. There was also a bug in checking its value on later use of it. Fixed.